Apps/Src/Task_VoltTempMonitor.c

```diff
@@ -0,0 +1,159 @@
+/* Copyright (c) 2020 UT Longhorn Racing Solar */
+#include <string.h>
+#include "Tasks.h"
+#include "os.h"
+#include "Voltage.h"
+#include "Temperature.h"
+#include "BSP_Fans.h"
+#include "CANbus.h"
+#include "Amps.h"
+#include "CAN_Queue.h"
+
+//declared in Tasks.c
+extern cell_asic Minions[NUM_MINIONS];
+
+void Task_VoltTempMonitor(void *p_arg) {
+    (void)p_arg;
+
+    OS_ERR err;
+
+    BSP_Fans_Init();
+    Voltage_Init(Minions);
+    Temperature_Init(Minions);
+
+    // SafetyCheck_Sem4 must only be signaled once per parameter at system boot up.
+    // These flags indicate was signaled for that parameter
+    bool voltageHasBeenChecked = false;
+    bool openWireHasBeenChecked = false;
+    bool temperatureHasBeenChecked = false;
+
+    CANData_t CanData;
+    CANPayload_t CanPayload;
+    CANMSG_t CanMsg;
+    while(1) {
+        // BLOCKING =====================
+        // Update Voltage Measurements
+        Voltage_UpdateMeasurements();
+        
+        // Check if voltage is NOT safe:
+        SafetyStatus voltageStatus = Voltage_CheckStatus();
+        if(voltageStatus != SAFE) {
+            if (voltageStatus == UNDERVOLTAGE) Fault_BitMap = Fault_UVOLT;
+            if (voltageStatus == OVERVOLTAGE) Fault_BitMap = Fault_OVOLT;
+            OSSemPost(&Fault_Sem4,
+                        OS_OPT_POST_1,
+                        &err);
+            assertOSError(err);
+        
+        } else if((voltageStatus == SAFE) && (!voltageHasBeenChecked)) {
+            // Signal to turn on contactor but only signal once
+            OSSemPost(&SafetyCheck_Sem4,
+                        OS_OPT_POST_1,
+                        &err);
+            assertOSError(err);
+
+            voltageHasBeenChecked = true;
+        }
+        //Send measurements to CAN queue
+        CanMsg.id = VOLT_DATA;
+        for (int i = 0; i < NUM_BATTERY_MODULES; i++){ //send all battery module voltage data
+            CanPayload.idx = i;
+            int voltage = Voltage_GetModuleMillivoltage(i);
+            CanData.w = voltage;
+            CanPayload.data = CanData;
+            CanMsg.payload = CanPayload;
+            CAN_Queue_Post(CanMsg);
+        }
+
+        // BLOCKING =====================
+        // Check if open wire is NOT safe:
+	
+	// TODO: get OpenWire to not take forever and actually run this
+        // SafetyStatus wireStatus = Voltage_OpenWire();
+	SafetyStatus wireStatus = SAFE;
```

Comment:
This is pretty bad. Need to figure out how we can actually run this efficiently.

Reply:
Yes. I opened #416 for this, but I don't think this should block us from merging this PR, since the fix will be pretty involved, and I think it can wait until after this is merged